tests/test_function.py

```diff
@@ -25,6 +25,34 @@ def test_get_function(execute):
     assert ft(a=5, p=3)["prod_p"] == 15  # New value for a, new value for p
 
 
+def test_use_cache(execute):
+    """
+    Test use_cache
+    """
+
+    art_code = """\n
+import lineapy
+a = 1
+p = 2
+lineapy.save(p, 'multiplier')
+b = a*p
+lineapy.save(b,'prod_p')
+"""
+    execute(art_code, snapshot=False)
+
+    code = """\n
+import lineapy
+ft = lineapy.get_function(['prod_p'], input_parameters=['a', 'p'], use_cache=['multiplier'])
+module_def = lineapy.get_module_definition(['prod_p'], input_parameters=['a', 'p'], use_cache=['multiplier'])
+assert ft()['prod_p'] == 2
+assert ft(a=5)['prod_p'] == 10
+assert ft(a=5, p=3)['prod_p'] == 10  # New value for a, cache value for p;  i.e., 5x2
+"""
+    res = execute(code, snapshot=False)
+    module_def = res.values["module_def"]
+    assert 'p = lineapy.get("multiplier", None).get_value()' in module_def
+
```

Comment:
Testing coverage is a larger matter here, but I think a basic thing we can do to increase coverage of the happy path is to have a version of this test with artifact version numbers specified and make sure it doesn't barf. 

Can we add that? should be simple and useful since we claim to support artifact versioning

Reply:
This is awesome, thank you!